operations/host_spawn.go

```diff
@@ -273,6 +293,46 @@ func hostModify() cli.Command {
 	}
 }
 
+func getPublicKey(ctx context.Context, client client.Communicator, keyFile, keyName string) (string, error) {
+	if keyFile != "" {
+		return readKeyFromFile(keyFile)
+	}
+
+	if keyName != "" {
+		return getUserKeyByName(ctx, client, keyName)
+	}
+
+	return "", nil
+}
+
+func readKeyFromFile(keyFile string) (string, error) {
+	if !utility.FileExists(keyFile) {
+		return "", errors.Errorf("key file '%s' does not exist", keyFile)
+	}
+
+	publicKeyBytes, err := ioutil.ReadFile(keyFile)
```

Comment:
Can we use `os.ReadFile` instead of `ioutil.ReadFile` since `ioutil` is deprecated? We might as well stop adding more `ioutil` code since it'll likely be gone soon.

Reply:
Ah, I didn't know that. 
On the other hand, it looks to me like `os.ReadFile` is new in go1.16. This being the case, maybe we should continue to use `ioutil.ReadFile` for now and bulk move later on once the team has all upgraded to the latest go?

Reply:
Evergreen has already compiled with go1.16 for many months now, so I don't think there's much value in waiting longer to use the latest features.

Reply:
done